### ISSUE TITLE
Switch to clerk for auth?

### ISSUE THREAD

**Issue by louisunlimited** [ParachuteTeam/Parachute#85]
Next Auth seems kinda unstable and I'm now questioning stability when pushed to production. We've sure all experienced setbacks when using it in dev, for ex this: 
<img width="327" alt="Screenshot 2023-07-19 at 13 17 38" src="https://github.com/ParachuteTeam/Parachute/assets/20884681/6ea46270-e218-4ef1-999c-f85d308a4ee9">

So here I propose using a third party auth service like [Clerk.dev](https://clerk.dev). It's free for < 5000 MAUs, which I think is pretty reasonable, and I've used it myself when building CS411 project and it seems reliable. 

@RaymondWHZ @HaiyueZhang @pastezzz Thoughs?

**Comment by RaymondWHZ** [comment]
The concept of this product is excellent. In this way, we can allow users to register by email directly as they wish, and maybe it can provide a better way to manage SSO providers and users.

The question is: how does it store user information? On its own platform? Besides, how should we change our current session feature and database adaptor?

**Comment by louisunlimited** [comment]
> The concept of this product is excellent. In this way, we can allow users to register by email directly as they wish, and maybe it can provide a better way to manage SSO providers and users.
> 
> The question is: how does it store user information? On its own platform? Besides, how should we change our current session feature and database adaptor?

Yes! We will let Clerk to handle all the storing/auth-ing of user info. We will need to remove all our db adapters and stuff form our current project since we don't need them anymore. So after this our db will be ONLY used for storing actual event logistics and such.

**Comment by RaymondWHZ** [comment]
> > The concept of this product is excellent. In this way, we can allow users to register by email directly as they wish, and maybe it can provide a better way to manage SSO providers and users.
> > The question is: how does it store user information? On its own platform? Besides, how should we change our current session feature and database adaptor?
> 
> @RaymondWHZ Yes! We will let Clerk to handle all the storing/auth-ing of user info. We will need to remove all our db adapters and stuff form our current project since we don't need them anymore. So after this our db will be ONLY used for storing actual event logistics and such.

Great! I think we can proceed with this idea.
